src/cpu/cgroup.rs

```diff
@@ -131,12 +137,61 @@ mod os {
                 _ => 0,
             };
 
-            if fields_encountered == CPU_SYS_NUMBER_OF_FIELDS {
+            if fields_encountered == CPU_SYS_V1_NUMBER_OF_FIELDS {
+                break;
+            }
+        }
+
+        if fields_encountered != CPU_SYS_V1_NUMBER_OF_FIELDS {
+            return Err(ProbeError::UnexpectedContent(
+                "Did not encounter all expected fields".to_owned(),
+            ));
+        }
+        let measurement = CgroupCpuMeasurement {
+            precise_time_ns: time,
+            stat: cpu,
+        };
+        Ok(measurement)
+    }
+
+    pub fn read_and_parse_v2_sys_stat(path: &Path) -> Result<CgroupCpuMeasurement> {
```

Comment:
It might make it a bit more manageable and readable if we split the v1 and v2 logic into their own separate files.
I don't think they share a lot of code.